### PR TITLE
[jit] Fix jit not round to even if const is folded

### LINKED ISSUE #40771: [jit] round not equivalent between python and torchscript
## 🐛 Bug

Originally reported by @vfdev-5  in https://github.com/pytorch/vision/pull/2342#discussion_r447587088

The behavior of `round` is not the same between torchscript and Python.
```python
import torch

def foo():    
    return round(2.5)

sfoo = torch.jit.script(foo)
print(foo(), sfoo())
# gives
# 2, 3.0
```
There are two differences: the value is different, but the returned type as well (float instead of int).

I'm using PyTorch version `'1.6.0.dev20200617'`

cc @suo @gmagogsfm

### PR DESCRIPTION
**Pull request by cloudhan**
Fixed #40771

### CODE REVIEW

**Comment by cloudhan** [comment]
@suo @gmagogsfm @fmassa would you mind looking at it?

File: torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp [hunk 1]
```diff
@@ -958,7 +958,9 @@ RegisterOperators reg2({
     DEFINE_INT_OP(aten::__lshift__, a << b),
     DEFINE_INT_OP(aten::__rshift__, a >> b),
 
-    DEFINE_UNARY_OP(aten::round, std::round(a), float, float),
+    // use this trick to enforce rounding to even.
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+    DEFINE_UNARY_OP(aten::round, std::round(a * 0.5) * 2.0, float, float),
```
Comment (gmagogsfm) [inline torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp:963]: How about the case say, where a = 0.8?

in python `round(0.8) = 1`
but `std::round(0.8*0.5) * 2 = 0`

Comment (cloudhan) [reply torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp]: should be good now

File: torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp [hunk 2]
```diff
@@ -688,6 +688,13 @@ void hashValue(Stack* stack) {
   push(stack, int64_t(hash));
 }
 
+// When a number is exactly halfway between two integers, use this trick to
```
Comment (gmagogsfm) [inline torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp:691]: Nit: explain that this is to match python bahavior

**Review by gmagogsfm — APPROVED** [review]

File: torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp (hunk 2 above)
Comment (cloudhan) [reply torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp]: done

File: torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp [hunk 3]
```diff
@@ -671,6 +671,19 @@ void hashValue(Stack* stack) {
   push(stack, int64_t(hash));
 }
 
+// As described in https://docs.python.org/3/library/functions.html#round
+// When a number is exactly halfway between two integers, python builtin round
+// function will round to even number. We use round(x/2)*2 to handle the
+// special halfway case. For positive 'x', round(x/2)*2 =
+// round((x_e + x_r)/2)*2 = x_e + round(x_r/2)*2, where x_e is an even integer,
+// x_r is either 0.5 of 1.5, round(x_r/2)*2 results a 0 or 2, so the final
+// result will always be a even number. Due to symmetric, it also applies to
```
Comment (gmagogsfm) [inline torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp:680]: s/symmetric/symmetricity

**Review by gmagogsfm — APPROVED** [review]

**Comment by gmagogsfm** [comment]
Thank you cloudhan for contributing this fix!

**Pull request closed without merging**